### PR TITLE
Additional reservation fields (TTVA-50)

### PR DESCRIPTION
**Pull request by jopesy**
Add `approved_at` and `requested_at` fields to the `Reservation` model. 
- `requested_at` is set when the reservation state changes
to `REQUESTED`.
- `approved_at` is set when the reservations is approved, i.e.
when the state is set from `REQUESTED` to `CONFIRMED` or
`WAITING_FOR_PAYMENT` (paid reservations).

This information is required so that we can modify the `expire_too_old_unpaid_orders` management command so that it doesn't cancel orders that are waiting for approval or will be paid with a payment link (Refs. TTVA-51, TTVA-49, TTVA-20).

Refs TTVA-50

### CODE REVIEW

File: resources/models/reservation.py [hunk 1]
```diff
@@ -364,12 +364,20 @@ def set_state(self, new_state, user):
...
             self.send_reservation_changed_mail_to_user()
             return
+        
+        if new_state == Reservation.REQUESTED:
+            self.requested_at = timezone.now()
+
+        if new_state in (Reservation.CONFIRMED, Reservation.WAITING_FOR_PAYMENT):
+            if old_state == Reservation.REQUESTED:
+                self.approver = user
+                self.approved_at = timezone.now()
```
Comment (danjacob-anders) [inline resources/models/reservation.py:374]: Is there a unit test covering these state changes?

Comment (jopesy) [reply resources/models/reservation.py:374]: Forgot to include them in the PR. I'll add one more test and add them to the PR.

Comment (jopesy) [reply resources/models/reservation.py:374]: Tests are included now

**Review by danjacob-anders — APPROVED** [review]

**Comment by danjacob-anders** [comment]
Looks great!

**Pull request merged**